src/libs/routes/errorHandler.ts

```diff
@@ -0,0 +1,15 @@
+import { Request, Response, NextFunction } from 'express';
+import { IError } from './interface';
+export default(err: IError, req: Request, res: Response, next: NextFunction) => {
+    if (res.headersSent) {
+        return next(err);
+      }
+    res.status(404).json(
```

Comment:
Don't hardcode 404 use error.code if it present( you pass from notFound middleware ) else use 500 if error.code is undefined